rust/src/main.rs

```diff
@@ -62,12 +70,12 @@ async fn cb(_ctx: &Context, event: EventType) {
 }
 
 #[async_std::main]
-async fn main() {
+async fn main() -> anyhow::Result<()> {
     let dbdir = current_dir().unwrap().join("deltachat-db");
     std::fs::create_dir_all(dbdir.clone()).unwrap();
```

Comment:
Import anyhow `Result` and `Context` trait with `use anyhow::{Context as _, Result}` at the top and then replace all `.unwrap()` calls with `.context("failed to create database directory")?` and similar. This works for all kinds of `Error`s and `Option`.

There is also an `unwrap` for `set_config` below and maybe others that can be replaced with `?`.